tests/dialects/test_stencil.py

```diff
@@ -560,3 +570,74 @@ def test_stencil_access():
     assert isinstance(access, AccessOp)
     assert access.offset == offset_index_attr
     assert access.temp.typ == temp_type
+
+
+# TODO: Move to a notebook at some point with proper documentation
+def test_1d3pt_stencil_construct():
+    """
+    An example 1d-3pt stencil implementation from XDSL Python-land
+    using the stencil dialect
+    """
+
+    shape = (8,)
+    space_order = 2
+    r = space_order // 2
+    symbolic_shape = (0 - r, shape[0] - r)
+
+    # Domain with halo shape
+    temp0 = TempType(len(shape), f32)
+
+    # Computational domain shape
+    field0 = FieldType([symbolic_shape], f32)
+
+    # The computation block
+    block0 = Block(arg_types=[temp0])
+
+    # Stencil computation
+    stencil_ops = [
+        stencil_acs_l := AccessOp.get(block0.args[0], (-1,)),
+        stencil_acs_c := AccessOp.get(block0.args[0], (0,)),
+        stencil_acs_r := AccessOp.get(block0.args[0], (1,)),
+        stencil_comp0 := Addf(stencil_acs_l, stencil_acs_c),
+        stencil_comp1 := Addf(stencil_comp0, stencil_acs_r),
+        # Define the return operation
+        return_op_ := ReturnOp.get([stencil_comp1]),  # type: ignore
+    ]
```

Comment:
I would suggest we use the implicit builder here?

Reply:
To build what?

Reply:
the whole module, I guess, would probably be more readable